File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
sum = 0
score_sum = 0;

for i in range(20):
  subject = input()
  score_info = subject.split(' ')
  if score_info[2] == 'P':
    continue
  score_sum += float(score_info[1])
  if score_info[2] == 'A+':
    sum += 4.5 * float(score_info[1])
  elif score_info[2] == 'A0':
    sum += 4.0 * float(score_info[1])
  elif score_info[2] == 'B+':
    sum += 3.5 * float(score_info[1])
  elif score_info[2] == 'B0':
    sum += 3.0 * float(score_info[1])
  elif score_info[2] == 'C+':
    sum += 2.5 * float(score_info[1])
  elif score_info[2] == 'C0':
    sum += 2.0 * float(score_info[1])
  elif score_info[2] == 'D+':
    sum += 1.5 * float(score_info[1])
  elif score_info[2] == 'D0':
    sum += 1.0 * float(score_info[1])
  elif score_info[2] == 'F':
    sum += 0.0 * float(score_info[1])
    
print(sum/score_sum)

File: 백준/Silver/25206. 너의 평점은/너의 평점은.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        InputStreamReader reader = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(reader);
        
        double sum = 0;
        double creditSum = 0;
        
        Main main = new Main();
        
        for(int i=0; i<20; i++) {
            String[] scoreInfo = br.readLine().split(" ");
            double credit = Double.parseDouble(scoreInfo[1]);
            String grade = scoreInfo[2];
            
            if(grade.equals("P")) {
                continue;
            }
            
            creditSum += credit;
            sum += (credit * main.gradeParser(grade));
        }
        
        System.out.println(sum/creditSum);
    }
    
    public double gradeParser(String grade) {
        if (grade.equals("A+")) {
            return 4.5;
        }else if (grade.equals("A0")) {
            return 4.0;
        }else if (grade.equals("B+")) {
            return 3.5;
        }else if (grade.equals("B0")) {
            return 3.0;
        }else if (grade.equals("C+")) {
            return 2.5;
        }else if (grade.equals("C0")) {
            return 2.0;
        }else if (grade.equals("D+")) {
            return 1.5;
        }else if (grade.equals("D0")) {
            return 1.0;
        }else if (grade.equals("F")) {
            return 0.0;
        }
        return 0.0;
    }
}